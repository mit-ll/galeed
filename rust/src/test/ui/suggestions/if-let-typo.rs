fn main() {
    let foo = Some(0);
    let bar = None;
    if Some(x) = foo {} //~ ERROR cannot find value `x` in this scope
    if Some(foo) = bar {} //~ ERROR mismatched types
    if 3 = foo {} //~ ERROR mismatched types
    //~^ ERROR mismatched types
    if Some(3) = foo {} //~ ERROR mismatched types
}
