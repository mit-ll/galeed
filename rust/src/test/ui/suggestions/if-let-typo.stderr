error[E0425]: cannot find value `x` in this scope
  --> $DIR/if-let-typo.rs:4:13
   |
LL |     if Some(x) = foo {}
   |             ^ not found in this scope
   |
help: you might have meant to use pattern matching
   |
LL |     if let Some(x) = foo {}
   |        ^^^

error[E0308]: mismatched types
  --> $DIR/if-let-typo.rs:5:8
   |
LL |     if Some(foo) = bar {}
   |        ^^^^^^^^^^^^^^^ expected `bool`, found `()`
   |
help: you might have meant to use pattern matching
   |
LL |     if let Some(foo) = bar {}
   |        ^^^
help: you might have meant to compare for equality
   |
LL |     if Some(foo) == bar {}
   |                  ^^

error[E0308]: mismatched types
  --> $DIR/if-let-typo.rs:6:12
   |
LL |     if 3 = foo {}
   |            ^^^ expected integer, found enum `Option`
   |
   = note: expected type `{integer}`
              found enum `Option<{integer}>`

error[E0308]: mismatched types
  --> $DIR/if-let-typo.rs:6:8
   |
LL |     if 3 = foo {}
   |        ^^^^^^^ expected `bool`, found `()`

error[E0308]: mismatched types
  --> $DIR/if-let-typo.rs:8:8
   |
LL |     if Some(3) = foo {}
   |        ^^^^^^^^^^^^^ expected `bool`, found `()`
   |
help: you might have meant to use pattern matching
   |
LL |     if let Some(3) = foo {}
   |        ^^^
help: you might have meant to compare for equality
   |
LL |     if Some(3) == foo {}
   |                ^^

error: aborting due to 5 previous errors

Some errors have detailed explanations: E0308, E0425.
For more information about an error, try `rustc --explain E0308`.
